selfdrive/ui/qt/widgets/input.cc

```diff
@@ -244,18 +244,35 @@ RichTextDialog::RichTextDialog(const QString &prompt_text, const QString &btn_te
   prompt->setStyleSheet("font-size: 42px; font-weight: light; color: #C9C9C9; margin: 45px;");
   main_layout->addWidget(new ScrollView(prompt, this), 1, Qt::AlignTop);
 
-  // confirm button
-  QPushButton* confirm_btn = new QPushButton(btn_text);
-  main_layout->addWidget(confirm_btn);
-  QObject::connect(confirm_btn, &QPushButton::clicked, this, &QDialog::accept);
+  // cancel + confirm buttons
+  QHBoxLayout *btn_layout = new QHBoxLayout();
+  btn_layout->setSpacing(30);
+  main_layout->addLayout(btn_layout);
+
+  if (cancel_text.length()) {
+    QPushButton* cancel_btn = new QPushButton(cancel_text);
+    btn_layout->addWidget(cancel_btn);
+    QObject::connect(cancel_btn, &QPushButton::clicked, this, &QDialog::reject);
+  }
+
+  if (confirm_text.length()) {
+    QPushButton* confirm_btn = new QPushButton(confirm_text);
+    btn_layout->addWidget(confirm_btn);
+    QObject::connect(confirm_btn, &QPushButton::clicked, this, &QDialog::accept);
+  }
 
   QVBoxLayout *outer_layout = new QVBoxLayout(this);
   outer_layout->setContentsMargins(100, 100, 100, 100);
   outer_layout->addWidget(container);
 }
 
 bool RichTextDialog::alert(const QString &prompt_text, QWidget *parent) {
-  auto d = RichTextDialog(prompt_text, tr("Ok"), parent);
+  auto d = RichTextDialog(prompt_text, tr("Ok"), "", parent);
+  return d.exec();
+}
+
+bool RichTextDialog::confirm(const QString &prompt_text, QWidget *parent) {
+  auto d = RichTextDialog(prompt_text, tr("Ok"), tr("Cancel"), parent);
   return d.exec();
 }
```

Comment:
what's this needed for? we already have `ConfirmationDialog`

Reply:
The RichTextDialog just had a better style. Doesn't seem simple to configure this per dialog
![Screenshot from 2022-11-02 15-52-14](https://user-images.githubusercontent.com/4038174/199617359-fb8ae503-23ee-49b1-ba9b-6079b9f25b9a.png)

Reply:
maybe we can pass a stylesheet to confirmationdialog

Reply:
big text medium text bool?

Reply:
i tried experimenting with some rounding on the confirmation dialog, it looked decent. i think it's the outlier with the multioptiondialog and richtextdialog adding rounding